core/quality/mysterium_morqa.go

```diff
@@ -18,45 +18,145 @@
 package quality
 
 import (
+	"bytes"
 	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"time"
 
 	log "github.com/cihub/seelog"
-	"github.com/mysteriumnetwork/node/requests"
+	"github.com/golang/protobuf/proto"
+	"github.com/mysteriumnetwork/metrics"
 )
 
 const (
 	mysteriumMorqaLogPrefix = "[Mysterium.morqa] "
+	mysteriumMorqaAgentName = "goclient-v0.1"
 )
 
+// HttpClient sends actual HTTP requests
+type HttpClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
 // MysteriumMORQA HTTP client for Mysterium QualityOracle - MORQA
 type MysteriumMORQA struct {
-	http                 requests.HTTPTransport
-	qualityOracleAddress string
+	http    HttpClient
+	baseURL string
 }
 
 // NewMorqaClient creates Mysterium Morqa client with a real communication
-func NewMorqaClient(qualityOracleAddress string) *MysteriumMORQA {
+func NewMorqaClient(baseURL string, timeout time.Duration) *MysteriumMORQA {
 	return &MysteriumMORQA{
-		requests.NewHTTPClient(1 * time.Minute),
-		qualityOracleAddress,
+		http:    &http.Client{Timeout: timeout},
+		baseURL: baseURL,
 	}
 }
 
 // ProposalsMetrics returns a list of proposals connection metrics
 func (m *MysteriumMORQA) ProposalsMetrics() []json.RawMessage {
-	req, err := requests.NewGetRequest(m.qualityOracleAddress, "proposals/quality", nil)
+	request, err := m.newRequestJSON(http.MethodGet, "proposals/quality", nil)
 	if err != nil {
 		log.Warn(mysteriumMorqaLogPrefix, "Failed to create proposals metrics request: ", err)
 		return nil
 	}
 
-	var metricsResponse ServiceMetricsResponse
-	err = m.http.DoRequestAndParseResponse(req, &metricsResponse)
+	response, err := m.http.Do(request)
 	if err != nil {
-		log.Warn(mysteriumMorqaLogPrefix, "Failed to request or parse proposals metrics: ", err)
+		log.Warn(mysteriumMorqaLogPrefix, err)
```

Comment:
Let's add some context to the logging, it could be hard to read logs without it.